modules/service/src/main/scala/ch/epfl/bluebrain/nexus/commons/service/persistence/IndexFailuresLog.scala

```diff
@@ -0,0 +1,68 @@
+package ch.epfl.bluebrain.nexus.commons.service.persistence
+
+import akka.actor.ActorSystem
+import akka.persistence.query.Offset
+import akka.stream.scaladsl.Source
+import io.circe.{Decoder, Encoder}
+
+import scala.concurrent.Future
+
+trait IndexFailuresLog {
+
+  /**
+    * @return an unique identifier for this failures log
+    */
+  def identifier: String
+
+  /**
+    * Records the failed event against this failures log.
+    *
+    * @param offset the offset to record
+    * @param event  the event to record
+    * @tparam T the generic type of the ''event''
+    * @return a future of [[Unit]] upon success or a failure otherwise
+    */
+  def storeEvent[T](offset: Offset, event: T)(implicit E: Encoder[T]): Future[Unit]
```

Comment:
I think we also need a persistent identifier to de-duplicate events with the same offset. It's quite possible to have two persistent identifiers with the same offset.